shotover/src/runner.rs

```diff
@@ -131,31 +122,42 @@ impl Shotover {
         let tracing = TracingState::new(config.main_log_level.as_str(), params.log_format)?;
         let runtime = Shotover::create_runtime(params.stack_size, params.core_threads);
 
-        // Log hot reload status
-        if params.hotreload {
-            tracing::info!(
-                "Hot reloading is ENABLED - shotover will support hot reload operations"
-            );
-        } else if let Some(socket_path) = &params.hotreload_from_socket {
-            tracing::info!(
-                "Hot reload CLIENT will request sockets from existing shotover at: {}",
-                socket_path
-            );
-        } else {
-            tracing::debug!("Hot reloading is disabled");
-        }
+        // Determine hot reload mode based on socket existence
+        let (hotreload_socket, hotreload_client) = match params.hotreload_socket {
```

Comment:
I suggest this change to simplify this logic:

<img width="726" height="1546" alt="Image" src="https://github.com/user-attachments/assets/8068cb62-dd89-442d-bc52-f3ce4dea0303" />

The changes include:
* remove `hotreload_client` field, instead derive it from `hotreload_socket` once we need the client.
* Move the `path::exists` check into the client constructor, this keeps the client logic in the client module and fits in well with the logic in run_inner.
* one downside of this approach is we lose the explicit `REPLACEMENT` mode vs `INITIAL mode` logs, but we already have plenty of logs walking through the hot reload process and I think its worth the simplification.